vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -472,15 +479,21 @@ def create_from_process_group(pg: ProcessGroup,
                 max_chunks=max_chunks,
             )
             handle = buffer_io.export_handle()
-            dist.broadcast_object_list([handle],
-                                       src=global_ranks[writer_rank],
-                                       group=pg)
+            if isinstance(pg, ProcessGroup):
+                dist.broadcast_object_list([handle],
+                                           src=global_ranks[writer_rank],
+                                           group=pg)
+            else:
+                pg.broadcast_obj(handle, writer_rank)
```

Comment:
What do you think about adding a class that exposes the same interface as `StatelessProcessGroup` and calls the `torch.distributed` methods under the hood?

It would be nice to smooth out these `if isinstance(pg, ProcessGroup) / else` statements everywhere

Reply:
A simpler option would just be a common `broadcast_obj` and `barrier` that handle the `ProcessGroup`/`StatelessProcessGroup` dispatching

Reply:
my plan is to fully transit to `StatelessProcessGroup` . you can see the `StatelessProcessGroup` code branch is much simpler.

Reply:
OK, What needs to happen in order to transition fully? I.E. how long do we expect the branches to be in place

Reply:
v1 is a good place to test the `StatelessProcessGroup`

Reply:
Sounds like it will still be around for a couple of months then? Probably worth factoring out just to clean up the code, but not a huge deal

Reply:
will do later if we find it necessary. right now the `StatelessProcessGroup` is still experimental.